packages/read-model/src/Product/Detail/ProductDetailViewElasticsearchFactory.php

```diff
@@ -0,0 +1,143 @@
+
+declare(strict_types=1);
+
+namespace Shopsys\ReadModelBundle\Product\Detail;
+
+use Shopsys\FrameworkBundle\Component\Utils\Utils;
+use Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser;
+use Shopsys\FrameworkBundle\Model\Product\Product;
+use Shopsys\FrameworkBundle\Model\Seo\SeoSettingFacade;
+use Shopsys\ReadModelBundle\Brand\BrandView;
+use Shopsys\ReadModelBundle\Brand\BrandViewFactory;
+use Shopsys\ReadModelBundle\Image\ImageView;
+use Shopsys\ReadModelBundle\Image\ImageViewFacadeInterface;
+use Shopsys\ReadModelBundle\Parameter\ParameterViewFactory;
+use Shopsys\ReadModelBundle\Product\Action\ProductActionViewFactory;
+use Shopsys\ReadModelBundle\Product\PriceFactory;
+
+class ProductDetailViewElasticsearchFactory
+{
+    /**
+     * @var \Shopsys\ReadModelBundle\Image\ImageViewFacadeInterface
+     */
+    protected $imageViewFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Seo\SeoSettingFacade
+     */
+    protected $seoSettingFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser
+     */
+    protected $currentCustomerUser;
+
+    /**
+     * @var \Shopsys\ReadModelBundle\Product\Action\ProductActionViewFactory
+     */
+    protected $productActionViewFactory;
+
+    /**
+     * @var \Shopsys\ReadModelBundle\Parameter\ParameterViewFactory
+     */
+    protected $parameterViewFactory;
+
+    /**
+     * @var \Shopsys\ReadModelBundle\Brand\BrandViewFactory
+     */
+    protected $brandViewFactory;
+
+    /**
+     * @param \Shopsys\ReadModelBundle\Image\ImageViewFacadeInterface $imageViewFacade
+     * @param \Shopsys\FrameworkBundle\Model\Seo\SeoSettingFacade $seoSettingFacade
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser $currentCustomerUser
+     * @param \Shopsys\ReadModelBundle\Product\Action\ProductActionViewFactory $productActionViewFactory
+     * @param \Shopsys\ReadModelBundle\Parameter\ParameterViewFactory $parameterViewFactory
+     * @param \Shopsys\ReadModelBundle\Brand\BrandViewFactory $brandViewFactory
+     */
+    public function __construct(
+        ImageViewFacadeInterface $imageViewFacade,
+        SeoSettingFacade $seoSettingFacade,
+        CurrentCustomerUser $currentCustomerUser,
+        ProductActionViewFactory $productActionViewFactory,
+        ParameterViewFactory $parameterViewFactory,
+        BrandViewFactory $brandViewFactory
+    ) {
+        $this->imageViewFacade = $imageViewFacade;
+        $this->seoSettingFacade = $seoSettingFacade;
+        $this->currentCustomerUser = $currentCustomerUser;
+        $this->productActionViewFactory = $productActionViewFactory;
+        $this->parameterViewFactory = $parameterViewFactory;
+        $this->brandViewFactory = $brandViewFactory;
+    }
+
+    /**
+     * @param array $product
+     * @return \Shopsys\ReadModelBundle\Product\Detail\ProductDetailView
+     */
+    public function createFromElasticsearchProduct(array $product): ProductDetailView
```

Comment:
`createFromArray`?

Reply:
changed in 27e1825cb8b7866cc0d1347f116fb21b2c085260